src/main/java/kg/jarkyn/cobspecserver/Runner.java

```diff
@@ -12,55 +12,64 @@
 
 import java.io.IOException;
 import java.net.ServerSocket;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
 
 public class Runner {
-    private static boolean KEEP_RUNNING = true;
-    private static int defaultPort      = 5000;
-    private static String publicResourcePath = "/Users/Jarkyn/Projects/8thLight/JAVA/cob_spec/public";
+    private static final int DEFAULT_PORT = 5000;
+    private static final boolean KEEP_RUNNING = true;
+    private static final String DOMAIN = "http://localhost";
+    private static final String PUBLIC_RESOURCE_PATH = "/Users/Jarkyn/Projects/8thLight/JAVA/cob_spec/public";
 
     public static void main(String[] args) {
         server().run();
     }
 
     private static Server server() {
-        return new Server(listener(), controller());
+        return new Server(executor(), listener(), controller());
+    }
+
+    private static ExecutorService executor() {
+        return Executors.newFixedThreadPool(10);
     }
 
     private static ListenerSocket listener() {
-        return new ListenerSocket(getServerSocket(), KEEP_RUNNING);
+        return new ListenerSocket(setupServerSocket(), KEEP_RUNNING);
```

Comment:
What you are suggesting was my initial implementation, and then I had feedback from two people that I should pass the flag in. I personally prefer what you are suggesting, but I obliged with the feedback received :(
I may put this back to how I had it originally, to me this seems quite ugly